### PR TITLE
GUI projectile homework

### PR DESCRIPTION
**Pull request by annafrei924**
Screenshot 2024-02-20 at 3 34 01 PM

### CODE REVIEW

File: src/main/java/freiman/projectile/ProjectileFrame.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        add(peakY);
+        add(interceptXLabel);
+        add(interceptX);
+        add(new JLabel());
+
+        add(calculateButton);
+
+        anglesSlider.addChangeListener(new ChangeListener() {
+            public void stateChanged(ChangeEvent ce) {
+                Projectile projectile = new Projectile(anglesSlider.getValue(), Double.parseDouble(velocityField.getText()));
```
Comment (andrewoid) [inline src/main/java/freiman/projectile/ProjectileFrame.java]: See how you copy/pasted all this code in **every single `Listener`**?
Can you reduce the code duplication?

**Comment by andrewoid** [comment]
Add a screenshot to your PR.

File: src/main/java/freiman/projectile/ProjectileFrame.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            @Override
+            public void update(DocumentEvent e) {
+                calculate(anglesSlider, velocityField, secondsField, x, y, peakY, interceptX);
+            }
+        });
+
+    }
+
+    private void calculate(JSlider anglesSlider, JTextField velocityField, JTextField secondsField,
+                           JLabel x, JLabel y, JLabel peakY, JLabel interceptX) {
```
Comment (andrewoid) [inline src/main/java/freiman/projectile/ProjectileFrame.java]: If you make these Components member variables then you don't need to pass them in this method.

**Comment by annafrei924** [comment]
Screenshot 2024-02-20 at 3 34 01 PM

**Comment by andrewoid** [comment]
> Screenshot 2024-02-20 at 3 34 01 PM
> 
> 

Add it to the description, not as a comment.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**